controller/src/beerocks/master/tasks/task.cpp

```diff
@@ -45,14 +45,11 @@ void task::response_received(std::string mac,
 
 void task::event_received(int event_type, void *obj)
 {
-    auto range = pending_events.equal_range(event_type);
-    for (auto it = range.first; it != range.second;) {
-        if (*it == event_type) {
-            it = pending_events.erase(it);
-            break;
-        } else {
-            it++;
-        }
+    auto it = pending_events.find(event_type);
+    if (it == pending_events.end()) {
+        TASK_LOG(DEBUG) << "received non-pending event " << event_type;
+    } else {
+        pending_events.erase(it);
```

Comment:
I think that originally the function was supposed to remove all the pending events on the range, so actually the the `break` statement is probably not in place. But I guess there is a good reason that it was added so your change seems ok.

Anyway, these commits have already merged a part of #659.